File: 2058-concatenation-of-array/2058-concatenation-of-array.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        int array_length=nums.length;

        int [] concatenated_array=new int [2*array_length];

        for(int index=0;index<array_length;index++){
            concatenated_array[index]=nums[index];
            concatenated_array[index+array_length]=nums[index];
        }
        return concatenated_array;
        
    }
}

File: 2058-concatenation-of-array/2058-concatenation-of-array.ts
function getConcatenation(nums: number[]): number[] {

    let array_length:number=nums.length

    let concatenated_array:number[]=new Array(2*array_length)
    // for(let index in nums){
    //     // console.log(index)
    //     // concatenated_array[index]=nums[index]
    //     // concatenated_array[index+array_length]=nums[index]
    //    console.log(typeof index)
    //    // String
    // }


     for(let index=0; index<array_length;index++){
        concatenated_array[index]=nums[index] 
        concatenated_array[index+array_length]=nums[index] 
     }
    
    return  concatenated_array;

    
};